### PR TITLE
User Settings Page

### LINKED ISSUE #59: [FE] User settings page
### Description
Users want to be able to view and change their account information. 

Create a new file `pages/users/account.jsx` that displays a user's account settings. This page should also allow users to edit their `name`.

- Get the current user id from the session
	- if there is no session (the user is not logged in) redirect them to `\login`
- use the `GET /api/users/:id` endpoint to retrieve data about the current user
- implement the user settings figma design
- clicking the Edit button should make `name` editable
	- make sure the name field isn't empty
	- on Save, a request should be sent to `PATCH /api/users/:id`
		- on success or failure, a toast should be displayed (use react-hot-toast)
	- on Cancel, the form should be reset and the display should revert back to the display view
- the **Deactivate** account button should update the user's role to "Inactive" (via a `PATCH`) and destroy their current session redirecting them to `/login`
    - however, before this occurs there should be a confirmation modal because this is an irreversible action
- Make the Account tab of the site navbar link to the page you created ( `/users/account`)
### Acceptance Criteria
- [ ] Proper error handling
- [ ] Implements Figma designs
- [ ] Proper JS doc comments above functions

### Other Notes
- N/a

### PR DESCRIPTION
**Pull request by siri-star**
## User Settings Page
Issue Number(s): #59 

What does this PR change and why?
Implements the account page, so users can view and update their personal information.

### Checklist
- [X] Requirements have been implemented
- [X] Acceptance criteria is met
- [X] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-a841a38d02db4428a7cdc85dcbb8a35c?pvs=4) have been updated or are not necessary
- [X] Code follows design and style guidelines
- [X] Commits follow guidelines (concise, squashed, etc)
- [X] Relevant reviewers (EM) have been assigned to this PR

### How To Test
Using the navigation bar, go to the account page by selecting the "account" tab.
(1) Ensure that the placeholder user's information is successfully retrieved and displayed.
(2) Ensure that the name can be changed to a non-empty string and a success toast is displayed. Otherwise, ensure that the name remains unchanged and an error toast is displayed.
(3) Ensure that once a user deactivates their account, their role is set to "Inactive" and that the webapp redirects to the login page.

### CODE REVIEW

File: src/pages/account.jsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+  return (
+    <Card cardStyle="mt-16 min-w-fit">
+      {showDeactivateModal ? (
+        <div>
+          <div className="fixed inset-0 bg-modal-background-gray opacity-60"></div>
+          <Card cardStyle="flex flex-col justify-between fixed z-10 inset-0 mt-32 mx-96 h-fit min-w-fit">
+            <h1>Deactivate {user.name}?</h1>
+            <p className="my-8">
+              Select “Confirm” to deactivate Jane and remove all access to the
```
Comment (afazio1) [inline src/pages/account.jsx]: change this from Jane to user's name

File: src/pages/account.jsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+  if (!data || data === undefined || !data.success) {
+    return <div>loading</div>;
+  }
+
+  const user = data.data;
+
+  return (
+    <Card cardStyle="mt-16 min-w-fit">
+      {showDeactivateModal ? (
```
Comment (afazio1) [inline src/pages/account.jsx]: possibly break the modal into its own component. You don't need to make a separate file for it though. Can keep it in this file

File: src/styles/globals.css [hunk 3]
```diff
@@ -25,7 +25,7 @@
   h1 {
     @apply text-3xl;
     @apply text-black;
-    @apply font-medium;
+    @apply font-bold;
```
Comment (afazio1) [inline src/styles/globals.css:28]: be careful of this affecting other pages

**Review by afazio1 — CHANGES_REQUESTED** [review]
looks good just a few minor things 👍

File: src/pages/account.jsx (hunk 2 above)
Comment (siri-star) [reply src/pages/account.jsx]: done!

File: src/styles/globals.css (hunk 3 above)
Comment (siri-star) [reply src/styles/globals.css:28]: I double checked figma and all the places where we use the h1 tag and it actually does need to be bolded everywhere :)

File: src/pages/account.jsx (hunk 1 above)
Comment (siri-star) [reply src/pages/account.jsx]: done!

**Review by afazio1 — APPROVED** [review]
good work 😎

**Pull request merged**